### PR TITLE
overlay: fallback without data only layers

### PR DESCRIPTION
**Pull request by giuseppe**
if the overlay data only layers feature is not available, then use a regular overlay lower layer.

The same functionality is already present in the mount helper for composefs.

### CODE REVIEW

File: drivers/overlay/overlay.go [hunk 1]
```diff
@@ -271,6 +272,18 @@ func (d *Driver) getSupportsVolatile() (bool, error) {
 	return supportsVolatile, nil
 }
 
+func (d *Driver) getSupportsDataOnly() (bool, error) {
```
Comment (rhatdan) [inline drivers/overlay/overlay.go:275]: nit: getSupportsDataOnly->supportsDataOnly

Comment (giuseppe) [reply drivers/overlay/overlay.go:275]: we already have a `supportsDataOnly *bool` in the `Driver` struct

**Comment by kwilczynski** [comment]
Related:

- https://github.com/cri-o/cri-o/pull/8619

**Comment by kwilczynski** [comment]
@giuseppe, would it be possible to expose this as a warning or a small helper so that users such as CRI-O could want the user that composefs is working in a degraded or unsupported (or not fully supported?) mode? We will users that might not have the kernel feature ready or backported for a while...

Unless you think handling this transparently is good enough.

**Comment by cgwalters** [comment]
Is there any reason we need to reimplement the mount logic here versus just calling out to `mount.composefs`? That would also ensure we automatically pick up other things like https://github.com/containers/composefs/pull/303/commits/91a304761c428551f96d7b6731b883955c7d96e1 (cosmetic, but useful imo).

**Review by cgwalters — APPROVED** [review]

**Comment by giuseppe** [comment]
> Is there any reason we need to reimplement the mount logic here versus just calling out to `mount.composefs`? That would also ensure we automatically pick up other things like [containers/composefs@91a3047](https://github.com/containers/composefs/commit/91a304761c428551f96d7b6731b883955c7d96e1) (cosmetic, but useful imo).

we are using a single overlay mount on top of multiple erofs mounts and basedirs, so there is not a single `IMAGE`

**Comment by cgwalters** [comment]
Ah right, so this relates to https://github.com/containers/storage/issues/2018

**Comment by kwilczynski** [comment]
**For posterity:**

This fixes an issue CRI-O had on platforms where the kernel lacked the required support.

**Pull request merged**